pkg/artifactory/resource/configuration/resource_artifactory_property_set_test.go

```diff
@@ -0,0 +1,102 @@
+package configuration_test
+
+import (
+	"fmt"
+	"github.com/go-resty/resty/v2"
+	"github.com/jfrog/terraform-provider-artifactory/v6/pkg/artifactory/resource/configuration"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/jfrog/terraform-provider-artifactory/v6/pkg/acctest"
+)
+
+func TestAccPropertySet(t *testing.T) {
+	const PropertySet = `
+resource "artifactory_property_set" "foo" {
+  name 		= "property-set1"
+  visible 	= true
+
+  property {
+      name = "set1property1"
+
+      predefined_value {
+        name 			= "passed-QA"
+        default_value 	= true
+      }
+
+      predefined_value {
+        name 			= "failed-QA"
+        default_value 	= false 
+      }
+
+      closed_predefined_values 	= true
+      multiple_choice 			= true
+  }
+
+  property {
+      name = "set1property2"
+    
+      predefined_value {
+        name 			= "passed-QA"
+        default_value 	= true
+      }
+
+      predefined_value {
+        name 			= "failed-QA"
+        default_value 	= false 
+      }
+
+      closed_predefined_values 	= false
+      multiple_choice 			= false
+  }
+}`
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { acctest.PreCheck(t) },
+		ProviderFactories: acctest.ProviderFactories,
+		CheckDestroy:      testAccPropertySetDestroy("foo"),
+
+		Steps: []resource.TestStep{
+			{
+				Config: PropertySet,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("artifactory_property_set.foo", "name", "property-set1"),
+					resource.TestCheckResourceAttr("artifactory_property_set.foo", "visible", "true"),
+					resource.TestCheckTypeSetElemAttr("artifactory_property_set.foo", "property.*.*", "set1property1"),
+					resource.TestCheckTypeSetElemAttr("artifactory_property_set.foo", "property.*.*", "set1property2"),
+					resource.TestCheckTypeSetElemAttr("artifactory_property_set.foo", "property.*.predefined_value.*.*", "passed-QA"),
+					resource.TestCheckTypeSetElemAttr("artifactory_property_set.foo", "property.*.predefined_value.*.*", "failed-QA"),
+				),
+			},
+		},
+	})
+}
+
```

Comment:
Please add tests to verify updating property set.

Reply:
Added, also refactored the tests.